File: backend/internal/model/message.go
package model

import "gorm.io/gorm"

type Message struct {
	gorm.Model

	ConversationID string `gorm:"size:64;not null;index" json:"conversation_id"`

	SenderID uint `gorm:"index;not null" json:"sender_id"`
	Sender   User `gorm:"foreignKey:SenderID" json:"sender"`

	ReceiverID uint `gorm:"index;not null" json:"receiver_id"`
	Receiver   User `gorm:"foreignKey:ReceiverID" json:"receiver"`

	Content string `gorm:"type:text;not null" json:"content"`
	IsRead  bool   `gorm:"not null;default:false" json:"is_read"`
}

File: frontend/src/types/message.ts
import type { PropertyHost } from "./property";

export type Message = {
  ID: number;
  CreatedAt: string;
  conversation_id: string;
  sender_id: number;
  sender: PropertyHost;
  receiver_id: number;
  receiver: PropertyHost;
  content: string;
  is_read: boolean;
};

export type ConversationSummary = {
  conversation_id: string;
  peer: PropertyHost & { phone?: string; role?: string; is_verified?: boolean };
  last_message: string;
  last_at: string;
  unread_count: number;
};

export type SendMessageInput = {
  receiver_id: number;
  content: string;
};
